File: dotnet-maui/ReservationsDotnetMaui/Models/Desk.cs
namespace ReservationsDotnetMaui.Models;

public class Desk
{
    public string Id { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public string OfficeId { get; set; } = string.Empty;
    public bool IsReserved { get; set; }
    public string? ReservedBy { get; set; }
    public DateTime? ReservedUntil { get; set; }
}

File: flutter/flutterapp/lib/models/desk.dart
class Desk {
  Desk({
    required this.id,
    required this.name,
    required this.officeId,
    this.isReserved = false,
    this.reservedBy,
    this.reservedUntil,
  });

  final String id;
  final String name;
  final String officeId;
  bool isReserved;
  String? reservedBy;
  DateTime? reservedUntil;
}
